File: src/main/java/com/example/sk/utils/sql/SqlUtils.java
package com.example.sk.utils.sql;

import org.jspecify.annotations.NonNull;
import org.springframework.core.io.ClassPathResource;
import org.springframework.util.StreamUtils;

import java.io.IOException;
import java.nio.charset.StandardCharsets;

public final class SqlUtils {

    private SqlUtils() {}

    public static @NonNull LoadSqlResult loadSql(@NonNull String classPathLocation) {
        final var resource = new ClassPathResource(classPathLocation);
        try {
            final var sqlString = StreamUtils.copyToString(resource.getInputStream(), StandardCharsets.UTF_8);
            return new LoadSqlSuccess(sqlString);
        } catch (IOException e) {
            return new LoadSqlError(e);
        }
    }
}

File: src/main/kotlin/com/example/sk/utils/sql/SqlUtils.kt
package com.example.sk.utils.sql

import org.springframework.core.io.ClassPathResource
import org.springframework.util.StreamUtils
import java.io.IOException
import java.nio.charset.StandardCharsets

object SqlUtils {
    fun loadSql(classPathLocation: String): LoadSqlResult {
        val resource = ClassPathResource(classPathLocation)
        return try {
            val sqlString = StreamUtils.copyToString(resource.getInputStream(), StandardCharsets.UTF_8)
            LoadSqlSuccess(sqlString)
        } catch (e: IOException) {
            LoadSqlError(e)
        }
    }
}